unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,237 @@
+
+    @staticmethod
+    def test_unregister(redis_client_auto_register):
+        """Check whether the redis.db.publish is called with correct arguments."""
+        redis_client_auto_register.redis_db.publish = MagicMock()
+        redis_client_auto_register.is_active = True
+        redis_client_auto_register.unregister(is_blocking=False)
+        transaction_id = redis_client_auto_register._blocking_command_responses[
+            "unregister"]["transaction_id"]
+        data = {"name": AREA_ID, "transaction_id": transaction_id}
+        redis_client_auto_register.redis_db.publish.assert_called_once_with(
+            f"{AREA_ID}/unregister_participant",
+            json.dumps(data))
+
+    @staticmethod
+    def test_unregister_is_active_false_throws_exception(redis_client_auto_register):
+        """Check if is active set to false throws exception when user is already unregistered."""
+        with pytest.raises(RedisAPIException,
+                           match="API is already unregistered from the market."):
+            redis_client_auto_register.is_active = False
+            redis_client_auto_register.unregister(is_blocking=True)
+
+    @staticmethod
+    @patch("d3a_api_client.redis_client_base.wait_until_timeout_blocking",
+           side_effect=AssertionError)
+    def test_unregister_is_active_true_throws_exception(
+            mock_wait_until_timeout_blocking,
+            redis_client_auto_register):
+        """Check if is active set to true throws exception."""
+        with pytest.raises(RedisAPIException,
+                           match="API unregister process timed out. "
+                                 "Server will continue processing "
+                                 "your request on the background and "
+                                 "will notify you as soon as "
+                                 "the unregistration has been completed."):
+            redis_client_auto_register.is_active = True
+            redis_client_auto_register.unregister(is_blocking=True)
+        mock_wait_until_timeout_blocking.assert_called()
+
+    @staticmethod
+    def test_on_register(redis_client_auto_register):
+        """Check the on_register function with correct message that doesn't throw exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID}
+        message = {"data": json.dumps(data)}
+        redis_client_auto_register._blocking_command_responses = {
+            "register": {"transaction_id": TRANSACTION_ID}}
+        redis_client_auto_register._on_register(message)
+        assert redis_client_auto_register.is_active is True
+        assert redis_client_auto_register.area_uuid == DEVICE_ID
+
+    @staticmethod
+    def test_on_unregister(redis_client_auto_register):
+        """Check the on_unregister function with correct message that doesn't throws exception."""
+        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID,
+                "response": "success"}
```

Comment:
Nit: no need to break it into two lines
```suggestion
        data = {"device_uuid": DEVICE_ID, "transaction_id": TRANSACTION_ID, "response": "success"}
```